cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java

```diff
@@ -178,4 +180,28 @@ private String generateMaxTerminationDateParam() {
       .withZone(ZoneId.systemDefault());
     return targetDate.format(formatter);
   }
+
+  private void validateConfiguration() throws MetricRetrievalException {
+    String host = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_HOST);
+    String typeKey = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.TYPE);
+    String maxRetriesKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.MAX_RETRIES);
```

Comment:
Shouldn't check them one by one here like this. You can just construct the RetryStrategy in the constructor fail there.

Reply:
I meant you don't need all these individual config checks... Just do a `this.retryStrategy = RertryStrategies.fromConfiguration(cConf)` in the constructor is good enough. If the error message is not clear enough about which config is wrong, update the `RertryStrategies.fromConfiguration` method appropriately in a generic way.

Reply:
Got it. Will do.

Reply:
Done